jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/cloudtrail/DenyPublicAccessS3BucketStoresCloudTrailLogs.java

```diff
@@ -0,0 +1,150 @@
+        String url = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        if (!StringUtils.isNullOrEmpty(url)) {
+            esUrl = url + "/aws/s3/_search";
+        }
+
+        String s3BucketName = resourceAttributes.get("s3bucketname");
+        String accountId = resourceAttributes.get(PacmanRuleConstants.ACCOUNTID);
+        boolean isPublicAccessEnabled = false;
+        if (!StringUtils.isNullOrEmpty(s3BucketName)) {
+            Map<String, Object> mustFilter = new HashMap<>();
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.NAME), s3BucketName);
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.ACCOUNTID),accountId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+        try {
+            isPublicAccessEnabled = checkForS3BucketCloudtrailLogsPublicAccess(esUrl, mustFilter);
+        } catch (Exception e) {
+            logger.error("unable to determine", e);
+            throw new RuleExecutionFailedExeption("unable to determine" + e);
+        }
+
+            if (isPublicAccessEnabled) {
+                List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+                Annotation annotation = null;
+                annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                annotation.put(PacmanSdkConstants.DESCRIPTION,"Ensure S3 bucket that stores cloudtrail Logs is not publicly accessible");
+                annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                annotation.put(PacmanRuleConstants.CATEGORY, category);
+                annotation.put(PacmanRuleConstants.NAME,resourceAttributes.get(PacmanRuleConstants.NAME));
+                issue.put(PacmanRuleConstants.VIOLATION_REASON,
+                        ruleParam.get(PacmanRuleConstants.RULE_ID) + " Violation Found!");
+                issueList.add(issue);
+                annotation.put(PacmanRuleConstants.ISSUE_DETAILS, issueList.toString());
+                logger.debug("S3 bucket that stores cloudtrail Logs is publicly accessible");
+                return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                        annotation);
+            }
+        }
+        logger.debug("S3 bucket that stores cloudtrail Logs is not publicly accessible");
+        logger.debug("========DenyPublicAccessS3BucketStoresCloudTrailLogs ended=========");
+        return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+
+    private boolean checkForS3BucketCloudtrailLogsPublicAccess(String esUrl, Map<String, Object> mustFilter) throws Exception {
+        logger.info("Validating the resource data from elastic search. ES URL:{}, FilterMap : {}", esUrl, mustFilter);
+        boolean validationResult = false;
```

Comment:
name it so it will be understandable what true and false mean. you got it correct in the main method above

Reply:
Addressed in commit